### DISCUSSION TITLE
tasmota 9.2.0 - Sonoff Basic R2 - AM2302 false negative temperature

### DISCUSSION THREAD

**Discussion by askan13** [arendst/Tasmota#10948] [Q&A] [answered]
Hello everybody!
++++Update 16.02.2021
Two deliverys of AM2302 are defect - no correct value under °C.
The third one works perfekt with all devices and temperature under 0°C.
+++++
I have 3 Sonoff Basic with Tasmota 9.2.0 and AM2302 in use.
Everything works great, but when the temperature is under 0°C then the temperature is incorrect.
When the temperature degrees to -0,3°C the webside of Sonoff shows for temperature −3276,4.
I get the same result on 3 Sonoff Basic with different AM2302.

I searched for this problem and found a discuss at [https://forum.arduino.cc/index.php?topic=129531.0](https://forum.arduino.cc/index.php?topic=129531.0)
It is possible to calculate the correct Value:
-0,3 = (3276,7−3276,4)*SGN(−3276,4)

Settings Sonoff:
Tasmota Version: 9.2.0 DE
All kinds of Logging are disabled.
AM2302 is connected to GPIO3 as AM2301.

Is it a known problem or is something done false by me?

Best regards stefan

**Comment by ascillato** [comment]
Please, update to latest Tasmota version from the development branch: http://ota.tasmota.com/tasmota/tasmota.bin.gz

**Reply by askan13** [reply]
Thanks for your fast answer.
Update done to v9.2.0.7
Reboot and currentless before new start.
Nothing changed, same result.
Stefan

**Comment by arendst** [comment, marked as answer]
It's a bit frustrating that my AM2302 sensor works just fine:

![image](https://user-images.githubusercontent.com/11044339/107874623-0c588680-6ebb-11eb-9760-548cbd4df817.png)

Anyway, I made a small change in the AM2301 temp calculation so pls try the dev version again.

**Reply by askan13** [reply]
Today i got an nodemcu and an Sonoff Dual R2 with tasmota v 9.2.0 DE from a friend with his AM2302.
This works great, also when the temperature under 0°C.
I dont know where i can find the failure, but it is definitely not tasmota, i am sorry.
I thought, when i test various devices with various AM2302 the failure is not my hardware. 
On Wednesday i can check my AM2302 sensors, they are from the same Chinese Seller.
Best regards.

**Comment by enricota66** [comment]
Hi everyone.
I have two nodemcu esp8266 with Tasmota 9.4 inside,  one DHT22 sensor wired on every board, 
 (setting of sensor is AM2301 template)
The humidity sensor seems to work fine, the temperature sensor work fine too but the data is flutter. see the pic below
![Schermata del 2021-05-31 07-44-05](https://user-images.githubusercontent.com/8460626/120147725-64579680-c1e7-11eb-80eb-f22da4c6f499.png)
The problems is the same on every Tasmota firmware version and not just 9.4.
There is someone have some idea to solve ??
Thanks in advance.

**Reply by sfromis** [reply]
"Wired on every board" sounds like you may be putting a temperature sensor too close to the "radiator". I always make sure to  keep some distance from a temperature sensor to the ESP8266, and avoid plausible heat conduction routes. Also ensure good stable power.

Anyway, Tasmota can only report what the sensor has to tell. Any solution has to be found outside of the Tasmota firmware.
